src/V3Trace.cpp

```diff
@@ -430,12 +430,14 @@ class TraceVisitor final : public EmitCBaseVisitor {
         AstAssign* const setterp = new AstAssign(fl, selectActivity(fl, code, VAccess::WRITE),
                                                  new AstConst(fl, AstConst::BitTrue()));
         if (AstCCall* const callp = VN_CAST(insertp, CCall)) {
-            callp->addNextHere(setterp);
+            callp->backp()->addNextHere(setterp);
```

Comment:
Previously, a CFunc used to be:
```
AstCFunc(
  AstCCall(...)
  AstCCall(...)
  AstNodeStmt(...)
)
```
I.e.: the CCall used to be in statement position.
Whith this patch CCall is an expression, and hence the CFunc looks like this:
```
AstCFunc(
  AstStmtMath(AstCCall(...))
  AstStmtMath(AstCCall(...))
  AstNodeStmt(...)
)
```
We need insert after the AstStmtMath (which is the `callp->backp()`), so we get this:
```
AstCFunc(
  AstStmtMath(AstCCall(...))
  ActivitySetterStmt()
  AstStmtMath(AstCCall(...))
  AstNodeStmt(...)
)
```

We know the backp is an `AstStmtMath`, because the only place a void function can be called is in an `AstStmtMath` (while non-void functions can be called anywhere subject to typing rules, inlcuding in `AstStmtMath`). Note currently basically all CFuncs have void return type, I'm expecting that to change in the future.

I will actually fix this instead so that `insertp` is the `AstStmtMath` rather than the CCall

Reply:
Done, see also the change of `visit(AstCCall*)` to `visit(AstStmtMath*)` to aid understanding